### DISCUSSION TITLE
Does MRTK3 pre 12 support Ultraleap Leap Motion Controller?

### DISCUSSION THREAD

**Discussion by Victor-x1** [microsoft/MixedRealityToolkit-Unity#11251] [Q&A] [answered]
The last Issue was posted was on August 31 but I'm confused if its currently supported. 
[Add Leap Motion UWP build support](https://github.com/microsoft/MixedRealityToolkit-Unity/issues/10038)

**Comment by keveleigh** [comment, marked as answer]
Yep! MRTK3 supports Ultraleap's Leap Motion controller via an OpenXR API layer: https://docs.ultraleap.com/openxr/

It should be included in any recent installation of the Ultraleap Tracking Software. After that, MRTK3 should "just work" in the default configuration when querying hand joint data from the OpenXR APIs and running a supported runtime (see the link above for more info on this). MRTK3 takes the hand joint data and synthesizes the rest of the input interactions that are needed, like pinch amount and the hand ray.

Please let me know if you run into any issues and I'm happy to help debug! I've tested it on the Windows Mixed Reality runtime specifically.

**Reply by Ivolutio** [reply]
Is there a way to get the Ultraleap hand-tracking in the Unity editor without an active OpenXR runtime? (Or is there a recommended mock runtime/simulator?)
I've been struggling to get Unity to see the Mixed Reality portal with simulator.
